### PR TITLE
Test case for mcg namespace supported S3 operations

### PR DESCRIPTION
**Pull request by parikshithb**
Signed-off-by: Parikshith <pbyregow@redhat.com>

### CODE REVIEW

**Review by ebenahar — APPROVED** [review]

File: ocs_ci/ocs/bucket_utils.py [hunk 1]
```diff
@@ -833,35 +833,45 @@ def s3_delete_bucket_website(s3_obj, bucketname):
...
         status (str): 'Enabled' or 'Suspended'. Default 'Enabled'
+        s3_client : Any s3 client resource
 
     Returns:
         dict : PutBucketVersioning response
     """
-    return s3_obj.s3_client.put_bucket_versioning(
-        Bucket=bucketname, VersioningConfiguration={"Status": status}
-    )
+    if s3_client:
```
Comment (romayalon) [inline ocs_ci/ocs/bucket_utils.py:849]: Versioning is out of scope for namespace buckets.

Comment (parikshithb) [reply ocs_ci/ocs/bucket_utils.py:849]: Ins't just list_object_versions supported on namespace buckets? Basically, I have enabled versioning on resource targets(aws here)  then uploaded object directly to the target and just list these objects using namespace bucket. Isn't this supported?

Comment (romayalon) [reply ocs_ci/ocs/bucket_utils.py:849]: ahh if you enabled versioning on the target bucket so it's ok, just notice that namespace buckets on top of Azure do not support list_object_versions.

Comment (parikshithb) [reply ocs_ci/ocs/bucket_utils.py:849]: Yup, will add a condition to skip object versioning if azure.

File: tests/e2e/lifecycle/mcg/test_mcg_namespace_s3_ops.py [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+        bucket_utils.s3_put_object(
+            s3_obj=mcg_obj, bucketname=bucket_name, object_key=obj_key, data=OBJ_DATA
+        )
+        object_keys.append(obj_key)
+        object_prefixes.append(f"{prefix}-{i}/")
+    mid_index = len(object_keys) // 2
+    return object_keys, object_prefixes, mid_index
+
+
+def get_list_and_verify(response, keys, verify="Contents", prefix="", delimiter=""):
```
Comment (fbalak) [inline tests/e2e/lifecycle/mcg/test_mcg_namespace_s3_ops.py:46]: Please add docstring with description what it does.

**Review by fbalak — COMMENTED** [review]
Looks great!
Would it make sense to split the `test_mcg_namespace_s3_ops` into more test cases? This way are not tested all operations if one of operations at the beginning fails...

File: tests/e2e/lifecycle/mcg/test_mcg_namespace_s3_ops.py (hunk 2 above)
Comment (parikshithb) [reply tests/e2e/lifecycle/mcg/test_mcg_namespace_s3_ops.py:49]: Added doc string

Comment (parikshithb) [reply tests/e2e/lifecycle/mcg/test_mcg_namespace_s3_ops.py:49]: Agree. Test is not split into 3 sub tests: Basic ops , object version and multipart.

**Review by fbalak — COMMENTED** [review]
Please add Polarion ids.

**Comment by parikshithb** [comment]
This test was created as single lifecycle testcase there are no subtest created for each. I have added lifecycle test polarion id above class.

**Review by fbalak — APPROVED** [review]
Ack

**Review by ebenahar — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**